sklearn/feature_selection/_rfe.py

```diff
@@ -187,11 +188,11 @@ def _fit(self, X, y, step_score=None):
         # and is used when implementing RFECV
         # self.scores_ will not be calculated when calling _fit through fit
 
-        tags = self._get_tags()
+        force_all_finite = not _safe_tags(self, key="allow_nan", default=True)
```

Comment:
`RFE` subclasses `BaseEstimator`, which would make it safe to call `self._get_tags`?

Reply:
Another thing, we had previously: `tags.get('allow_nan', True)`. So `get` was useless as well, isn't it? The tag should always be `False` due to the default?

Regarding calling `_safe_tags`, it would not be detrimental to always call `_safe_tags` and it would be consistent.

Reply:
> So get was useless as well, isn't it?

I think so yeah

> Regarding calling _safe_tags, it would not be detrimental to always call _safe_tags and it would be consistent.

I agree with @thomasjpfan and I'd prefer to always use `self._get_tags()` and `_safe_tags(self.base_estimator_)`. In particular because passing `default=...` to `_safe_tags` on a builtin estimator makes no sense, as you just noted

Reply:
OK works with me.